pyfar/io/io.py

```diff
@@ -490,6 +492,356 @@ def default_audio_subtype(format):
+
+    # final data shape
+    final_shape = [n_nodes, len(expressions)]
+    for key in parameters:
+        final_shape.append(len(parameters[key]))
+    final_shape.append(len(domain_data))
+
+    # temporary shape
+    n_combinations = np.prod(final_shape[2:-1]) if parameters else 1
+    temp_shape = [n_nodes, len(expressions), n_combinations, len(domain_data)]
+
+    # create pairs of parameter values
+    pairs = np.meshgrid(*[x for x in parameters.values()])
+    parameter_pairs = dict()
+    for idx, key in enumerate(parameters):
+        parameter_pairs[key] = pairs[idx].T.flatten()
+
+    # loop over expressions, domain, parameters
+    # extract the data by comparing with header
+    # first fill the array with temporary shape, then reshape
+    data_in = raw_data[:, -n_entries:]
+    data_out = np.full(temp_shape, np.nan, dtype=dtype)
+    for expression_idx, expression_key in enumerate(expressions):
+        expression_mask = expressions_header == expression_key
+        for parameter_idx in range(temp_shape[2]):
+            parameter_mask = np.full_like(expression_mask, True)
+            for key in parameters:
+                parameter_mask &= parameter_header[key] \
+                    == parameter_pairs[key][parameter_idx]
+            for domain_idx, domain_value in enumerate(domain_data):
+                domain_mask = domain_header == domain_value
+                mask = parameter_mask & domain_mask & expression_mask
+                if any(mask):
+                    data_out[:, expression_idx, parameter_idx, domain_idx] \
+                        = data_in[:, mask].flatten()
+                else:
+                    if parameters == all_parameters:
+                        warnings.warn(r'Parameter data is inconsistent. \
+                            Missing data is filled with nans.')
+
+    # reshape data to final shape
+    data_out = np.reshape(data_out, final_shape)
+
+    # create object
+    comment = ', '.join(' '.join(x) for x in zip(all_expressions, units))
+    if domain == 'freq':
+        data = FrequencyData(
+            data_out, domain_data, dtype=dtype, comment=comment)
+    else:
+        data = TimeData(data_out, domain_data, comment=comment)
+
+    # create coordinates
+    if n_dimension > 0:
+        coords_data = np.real(raw_data[:, 0:n_dimension])
+        x = coords_data[:, 0]
+        y = coords_data[:, 1] if n_dimension > 1 else np.zeros_like(x)
+        z = coords_data[:, 2] if n_dimension > 2 else np.zeros_like(x)
+        coordinates = Coordinates(x, y, z)
+    else:
+        coordinates = None
+    return data, coordinates
+
+
+def read_comsol_header(filename):
+    """Returns header information from exported data from COMSOL.
+
+    Parameters
+    ----------
+    filename : str, Path
+        Input file. Excepted input files are '.txt', '.dat' and '.csv'. '.csv'
+        Files are strongly recommended, since .txt and .dat are varies in its
+        format definition.
+
+    Returns
```

Comment:
These docstrings could be a little more detailed. What about explaining the meanings and corresponding options of `Expressions`, `Parameters`, `domain` and `domain_data` in Comsol first and and then describing the format of the returned data incl. the short examples?